src/components/fx/layout_attributes.js

```diff
@@ -57,11 +57,18 @@ module.exports = {
             color: extendFlat({}, fontAttrs.color)
         },
         namelength: {
-            valType: 'number',
-            min: 0,
+            valType: 'integer',
+            min: -1,
             dflt: 15,
```

Comment:
I kept only the global default, because the trace-level default [inherits](https://github.com/plotly/plotly.js/pull/1822/files#diff-2d7db1bff018d6a0f823bed5c4994b69R18) from the global so doesn't use a default of its own.

You're right that we want a default - in fact, we *need* a default of 15 to preserve backward compatibility - this is just how that default gets propagated.